tools/code-generation/smithy/codegen/cpp-smoke-tests-codegen/src/main/java/com/amazonaws/util/awsclientsmithygenerator/generators/CppSymbolVisitor.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+ package com.amazonaws.util.awsclientsmithygenerator.generators;
+
+ import software.amazon.smithy.codegen.core.*;
+ import software.amazon.smithy.model.Model;
+ import software.amazon.smithy.model.shapes.*;
+ import software.amazon.smithy.utils.StringUtils;
+ 
+ import java.util.logging.Logger;
+ 
+ import static java.lang.String.format;
+ 
+ /**
+  * For a list of simple types see <a href = https://smithy.io/2.0/spec/simple-types.html#simple-types>Smithy Simple Types</a>
+  * For a list of aggregate types see <a href = https://smithy.io/2.0/spec/aggregate-types.html#>Aggregate Types</a>
+  */
+ 
+ public class CppSymbolVisitor implements SymbolProvider, ShapeVisitor<Symbol> {
+ 
+     private static final Logger LOGGER = Logger.getLogger(CppSymbolVisitor.class.getName());
+     private final Model model;
+
+     public CppSymbolVisitor(Model model) {
+         this.model = model;
+     }
+ 
+     /**
+      * @param shape The smithy shape
+      * @return Returns the C++ Symbol
+      */
+     @Override
+     public Symbol toSymbol(Shape shape) {
+         Symbol symbol = shape.accept(this);
+         LOGGER.info(() -> format("Creating symbol from %s: %s", shape, symbol));
+         return symbol;
+     }
+ 
+     @Override
+     public String toMemberName(MemberShape shape) {
+         return StringUtils.capitalize(SymbolProvider.super.toMemberName(shape));
+     }
+ 
+     @Override
+     public Symbol blobShape(BlobShape blobShape) {
+         return createSymbolBuilder(blobShape, "stringstream").build();
```

Comment:
need top use Aws::StringStream not std string stream

Reply:
ack